featuretools/synthesis/utils.py

```diff
@@ -59,7 +59,7 @@ def get_unused_primitives(specified, used):
 def _schemas_equal(feature_schema, primitive_schema):
     """Checks if a feature schema matches a primitive schema"""
     if primitive_schema.logical_type and \
-            primitive_schema.logical_type != feature_schema.logical_type:
+            not isinstance(feature_schema.logical_type, type(primitive_schema.logical_type)):
```

Comment:
could this potentially cause issues comparing two ordinals with different orders?

Reply:
Ah - the issue that caused me to change this was involved comparing two datetimes that had different `datetime_format` values. A `Datetime` column with a format was not found to be a match for a primitive that had a `Datetime` input because of the format value difference.

I think the idea here is that if we define a primitive input schema as `ColumnSchema(logical_type=Ordinal)` that we want it to match a column that might have a logical type instance of `Ordinal` with order values defined. The previous check of `primitive_schema.logical_type != feature_schema.logical_type` would fail to find a match if one had order values defined and the other did not.

So having said that I, think we want to match on logical type alone and not specific attributes of the instantiated logical types?

Reply:
Maybe the name `_schemas_equal` isn't quite right given that behavior?

Reply:
yeah.  the logic of this method seems practically identitacal to `is_valid_input`, should we just have one method?

Reply:
Seems like we should be able to have a single method. I'll update.

Reply:
Replaced `_schemas_equal` with `is_valid_input` and updated associated tests.